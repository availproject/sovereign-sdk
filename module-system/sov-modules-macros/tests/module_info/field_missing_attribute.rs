use sov_modules_api::{Context, ModuleInfo};
use sov_state::StateMap;

#[derive(ModuleInfo)]
struct TestStruct<C: Context> {
    #[address]
    pub address: C::Address,

    test_state1: StateMap<u32, u32>,

    #[state]
    test_state2: StateMap<Vec<u8>, u64>,
}

fn main() {}
