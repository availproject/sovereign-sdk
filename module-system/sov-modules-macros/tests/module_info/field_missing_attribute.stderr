error: This field is missing an attribute: add `#[module]`, `#[state]` or `#[address]`.
 --> tests/module_info/field_missing_attribute.rs:9:5
  |
9 |     test_state1: StateMap<u32, u32>,
  |     ^^^^^^^^^^^
